src/coreclr/vm/readytoruninfo.cpp

```diff
@@ -1771,5 +1789,161 @@ ModuleBase* CreateNativeManifestModule(LoaderAllocator* pLoaderAllocator, IMDInt
     void *mem = pamTracker->Track(pLoaderAllocator->GetLowFrequencyHeap()->AllocMem(S_SIZE_T(sizeof(NativeManifestModule))));
     return new (mem) NativeManifestModule(pLoaderAllocator, pManifestMetadata, pModule, pamTracker);
 }
-
 #endif // DACCESS_COMPILE
+
+ReadyToRun_EnclosingTypeMap ReadyToRun_EnclosingTypeMap::EmptyInstance;
+ReadyToRun_TypeGenericInfoMap ReadyToRun_TypeGenericInfoMap::EmptyInstance;
+ReadyToRun_MethodIsGenericMap ReadyToRun_MethodIsGenericMap::EmptyInstance;
+
+mdTypeDef ReadyToRun_EnclosingTypeMap::GetEnclosingType(mdTypeDef input, IMDInternalImport* pImport)
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > TypeCount) || (rid == 0))
+    {
+        mdTypeDef enclosingType;
+        HRESULT hr = pImport->GetNestedClassProps(input, &enclosingType);
+
+        if (FAILED(hr))
+        {
+            if (hr == CLDB_E_RECORD_NOTFOUND)
+                return mdTypeDefNil;
+            ThrowHR(hr);
+        }
+
+        return enclosingType;
+    }
+    return TokenFromRid((&TypeCount)[rid], mdtTypeDef);
+}
+
+HRESULT ReadyToRun_EnclosingTypeMap::GetEnclosingTypeNoThrow(mdTypeDef input, mdTypeDef *pEnclosingType, IMDInternalImport* pImport)
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > TypeCount) || (rid == 0))
+    {
+        return pImport->GetNestedClassProps(input, pEnclosingType);
+    }
+
+    *pEnclosingType = TokenFromRid((&TypeCount)[rid], mdtTypeDef);
+
+    if (*pEnclosingType == mdTypeDefNil)
+    {
+        return CLDB_E_RECORD_NOTFOUND;
+    }
+    return  S_OK;
+}
+
+ReadyToRunTypeGenericInfo ReadyToRun_TypeGenericInfoMap::GetTypeGenericInfo(mdTypeDef input, bool *foundResult)
+{
+    uint32_t rid = RidFromToken(input);
+    if ((rid > TypeCount) || (rid == 0))
+    {
+        *foundResult = false;
+        return (ReadyToRunTypeGenericInfo)0;
+    }
+    uint32_t index = rid - 1;
+    uint8_t *pTypeGenericInfo = ((uint8_t*)&TypeCount) + sizeof(uint32_t);
+    uint8_t entry = pTypeGenericInfo[index / 2];
+    if (!(index & 1))
+    {
+        entry >>= 4;
+    }
```

Comment:
Where is this data layout documented?

Reply:
Nowhere, yet. That was next on my list of things to do once I had my final perf numbers to put into the checkin comment, and proved that it worked fairly completely. I should have that data by end of night tonight, and then I'll work on fixing up the command line for controlling this stuff, and adding some docs.